### PR TITLE
April UI Maintenance: Mitigate firebase apk pushing issue

### PR DESCRIPTION
**Pull request by fadidurah**
April UI Maintenance: Mitigate firebase apk pushing issue.

We use local app installation to enable testing RC apks as well as providing a faster, more consistent apk installation process as opposed to using playstore (internet connection on the devices is pretty slow, which causes increase in cost if we use playstore). 

We ran into an issue where firebase was no longer recieving these local apks in the /data/local/tmp folder that we have been using up to this point to store our apks for installaiton. The solution was to push the files into the /sdcard/ folder instead, and copy them over to /data/local/tmp.

### CODE REVIEW

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/CopyFileRule.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package com.microsoft.identity.client.ui.automation.rules;
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/CopyFileRule.java:23]: LICENSE

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/CopyFileRule.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import com.microsoft.identity.client.ui.automation.broker.BrokerMicrosoftAuthenticator;
+import com.microsoft.identity.client.ui.automation.browser.BrowserEdge;
+import com.microsoft.identity.client.ui.automation.logging.Logger;
+import com.microsoft.identity.client.ui.automation.utils.AdbShellUtils;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class CopyFileRule implements TestRule {
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/CopyFileRule.java:47]: javadoc

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/CopyFileRule.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            OutlookApp.OUTLOOK_APK,
+            WordApp.WORD_APK,
+            BrowserEdge.EDGE_APK,
+            BrokerLTW.BROKER_LTW_APK,
+            BrokerLTW.OLD_BROKER_LTW_APK
+    };
+
+    public CopyFileRule() {
+        mSourceFolder = "/sdcard/";
+        mDestFolder = "/data/local/tmp/";
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/CopyFileRule.java]: Can we define `/sdcard` and `/data/local/tmp` in private static final at top of this class

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/CopyFileRule.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Override
+    public Statement apply(final Statement base, final Description description) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                Logger.i(TAG, "Applying rule....");
+                Logger.i(TAG, "Copying into /data/local/tmp...");
+
+                for (String apkFileName: mApkFileNames){
```
Comment (shahzaibj) [inline uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/CopyFileRule.java]:
```suggestion
                for (final String apkFileName: mApkFileNames){
```

**Review by shahzaibj — APPROVED** [review]

File: uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/CopyFileRule.java (hunk 3 above)
Comment (fadidurah) [reply uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/CopyFileRule.java]: Sure thing

**Review by rpdome — APPROVED** [review]

**Pull request merged**